### ISSUE TITLE
Frontend implementation for URI Mathesar type

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#256]
## Requirements

Users should be able to:
- view data in columns that have the Mathesar URI custom type
- see validation errors if the data they enter is not a valid URI
- change data type to and from the URI DB type
- view, set, and change data type options related to URI data type
- ~~group data by URI-specific grouping options~~
- ~~filter data by URI-specific filtering options~~

Update:
- Grouping will be handled in #1073 
- Filtering will be handled in #1072 

## DB Type Mapping
All columns of this type map to `MATHESAR_TYPES.URI`. Please note that we should make sure it's available before showing it as an available type.

### Design
- The [Global Data Types](https://wiki.mathesar.org/design/specs/global-data-type-components) spec has the latest version of the UX.
- The [URL Data Types](https://wiki.mathesar.org/design/specs/url-data-type) spec has an older version of the UX but more information on user scenarios.

### Backend
- `/api/v0/tables/` lists the name and DB type of each columns 
- `/api/v0/tables/<id/columns/` lists column information in more detail
- `/api/v0/databases/<id>/types/` maps Mathesar types to DB types and provides a list of filtering and grouping options per-type.
  - These should be used to populate the filter and grouping options dynamically. 
  - This should also be used to determine the availability of the URI type on a given database. It is a custom type so it's not available on all databases.
- Updates to column types and type options should happen via `PATCH` requests to `/api/v0/tables/<id>/columns/<id>/`. These updates include
  - Changes to the data type (`type`)

## Additional Context
- #236 (design)
- #269 (concept of Mathesar types)
- #498 (conversation about data types)
- Blocked by backend work:
  - #412

**Comment by kgodey** [comment]
Marking this as ready since filtering and grouping have been moved to other issues and remaining backend work is done.

**Comment by seancolsen** [comment]
I'm closing in favor of the following more specific tickets which should address all the remaining work needed here.

- #1322
- #1276
- #1073 
- #1072
